### PR TITLE
Feature: Decimal data type(not optimized)

### PR DESCRIPTION
**Pull request by broccoliSpicy**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #1867 

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Comment by iamlinjunhong** [comment]
Unit test of parser should be added to mysql_sql_test.go

**Review by zzl200012 — APPROVED** [review]
Storage part LGTM.

File: pkg/vectorize/ne/ne.go [hunk 1]
```diff
@@ -1825,7 +1825,7 @@ func decimal128NeScalar(x types.Decimal128, ys []types.Decimal128, xScale, yScal
 			scaleDiff := xScale - yScale
 			ysScaled[i] = y
 			// since the possible scale difference is (0, 38], and 10**38 can not fit in a int64, double loop is necessary
-			for i := 0; i < int(scaleDiff); i++ {
+			for j := 0; j < int(scaleDiff); j++ {
```
[the remark this answers is not in the document]
Comment (broccoliSpicy) [reply pkg/vectorize/ne/ne.go:1828]: no, the go compiler won't by default give warnings about variable shadowing, but it did provide a additional tool for shadowed variable checking, [go tool shadow](https://pkg.go.dev/golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow)

variable shadowing is quite ubiquitous in golangish error handling code like this, 
```go
func foo() error {
    var err error
    if value, err := bar(); err != nil {
         ....
    }
}
```


I never had a comprehensive grasp of the go tool chain and it already bit me numerous times...

**Comment by nnsgmsone** [comment]
lgtm

File: pkg/frontend/mysql_cmd_executor.go [hunk 2]
```diff
@@ -429,6 +429,34 @@ func getDataFromPipeline(obj interface{}, bat *batch.Batch) error {
 						row[i] = vs[rowIndex]
 					}
 				}
+			case types.T_decimal64:
+				fmt.Println("the scale for this decimal is", vec.Typ.Scale)
```
Comment (m-schen) [inline pkg/frontend/mysql_cmd_executor.go:433]: Is this code left over from the development process?

**Review by m-schen — COMMENTED** [review]
I think some unit tests are needed to ensure that
decimal related functions in decimal.go work correctly.

e.g. TestCompareDecimal128Decimal128

File: pkg/frontend/mysql_cmd_executor.go (hunk 2 above)
Comment (broccoliSpicy) [reply pkg/frontend/mysql_cmd_executor.go]: solved.

**Review by jianwan0214 — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Pull request merged**